src/Compilers/CSharp/Portable/Symbols/Source/SourceParameterSymbol.cs

```diff
@@ -54,42 +55,32 @@ internal abstract class SourceParameterSymbol : SourceParameterSymbolBase
             }
 
             ImmutableArray<CustomModifier> inModifiers = ParameterHelpers.ConditionallyCreateInModifiers(refKind, addRefReadOnlyModifier, context, declarationDiagnostics, syntax);
-
-            if (!inModifiers.IsDefaultOrEmpty)
-            {
-                return new SourceComplexParameterSymbolWithCustomModifiersPrecedingByRef(
-                    owner,
-                    ordinal,
-                    parameterType,
-                    refKind,
-                    inModifiers,
-                    name,
-                    locations,
-                    syntax.GetReference(),
-                    isParams,
-                    isExtensionMethodThis);
-            }
+            Debug.Assert(!inModifiers.IsDefault);
 
             if (!isParams &&
                 !isExtensionMethodThis &&
+                scope == DeclarationScope.Unscoped &&
```

Comment:
>scope == DeclarationScope.Unscoped && [](http://example.com/codeflow?start=16&length=37)

I do not think addition of this condition is justified. We should still be able to create ```SourceSimpleParameterSymbol``` in this case. ```SourceComplexParameterSymbol``` doesn't add any interesting behavior around ```DeclarationScope``` that ```SourceSimpleParameterSymbol``` cannot handle, other than the storage, of course. As I suggested below, we should simply move the storage to this class and restore the name for ```SourceComplexParameterSymbolWithCustomModifiersPrecedingByRef``` class.